chat-app/src/MessageForm.js

```diff
@@ -0,0 +1,47 @@
+// import 'dotenv/config';
+import { useState } from 'react';
+
+export const MessageForm = ({ messageText, setMessageText, sendMessage }) => {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(null);
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    setLoading(true);
+    setError(null);
+
+    try {
+      await sendMessage();
+    } catch (err) {
+      setError("Failed to send message");
```

Comment:
Consider providing more detailed error messages to help users understand why sending a message failed. This can improve user experience.